kadmin/load.c

```diff
@@ -332,7 +332,7 @@ parse_extensions(char **strp, HDB_extensions **e)
 	void *d;
 
 	len = strlen(p);
-	d = emalloc(len);
+	d = emalloc(len+1);
```

Comment:
Are you sure this is right?

Reply:
Coverity's position is 
"
   10. strlen_assign: Setting variable len to the return value of strlen called with argument p.
334        len = strlen(p);
   CID 12491 (#1 of 1): Out-of-bounds access (OVERRUN)11. alloc_strlen: Allocating insufficient memory for the terminating null of the string.
335        d = emalloc(len);
336"

I don't see what it's complaining about, and seems like it's a false positive.

Reply:
Need to add 1 for the null terminator. A char string of 'len' is 4 bytes, because it's 'l e n \0'

But strlen( 'len' ) just returns 3. So making a place to copy it to needs to be 1 larger for the \0. Otherwise it can lead to overread or out of bounds access.

Sorry for sounding like a 'C' class teacher. Got bit by this last month.

Reply:
Your observation would be correct for `hex_encode()`. 

However `hex_decode()` is decoding binary information from a `NUL` terminated hex string. It returns binary data and a length. It should not need to be terminated.

Reply:
`hex_decode()` should really be called with `len / 2`.

Reply:
My apologies. Just like me, Coverity didn't look into the decode function to realize the resultant usage of the allocated memory didn't need to be null terminated.